### DISCUSSION TITLE
Feature Density Histograms

### DISCUSSION THREAD

**Discussion by katemortensen** [GMOD/jbrowse-components#1348] [General] [answered]
Hello! I'm new to Jbrowse and the computer world in general, so bear with me while I try to describe my question. I'm trying to personalize my genome browser. Specifically, I want a feature density histogram to appear when I'm zoomed-out far enough. 

I've tried editing the track.conf file as described in the jbrowse tutorial [https://jbrowse.org/docs/alignments.html](https://jbrowse.org/docs/alignments.html). No luck.

In desperation, I've also tried editing the trackList.json to include a histogram object in an attempt to copy what I've seen in other genome browsers. I'm not confident this is the correct approach. No luck.

At this point, I don't know what question I should be asking.

Any help would be much appreciated!

**Comment by cmdcolin** [comment]
Hi there
Good question

Technically this is the new jbrowse 2 discussion forum (just launched), we don't yet have a jbrowse 1 discussion forum just the issue thread so I'll just go ahead here :)


It does generally involves hand-editing to add the histograms config, except if using bin/add-track-json.pl which just adds chunks of json to your config file [1]

You may have found this section
https://jbrowse.org/docs/alignments.html#histograms-configuration

This shows how to add the histograms to a track in tracks.conf format. This can be done for any Alignments2 track or any CanvasFeatures track also. If you see in your trackList.json that your track type is Alignments2 or CanvasFeatures then you can use the histograms config.

Here is an example histograms config for a BAM track

```
[tracks.bam]
urlTemplate=alignments.bam
histograms.storeClass=JBrowse/Store/SeqFeature/BigWig
histograms.urlTemplate=coverage.bw
```


In trackList.json format, we can represent the same concept equivalently like this (tracks.conf and trackList.json are two different syntaxes for making track configs)


```
{
  "tracks": [
    {
      "label": "bam",
      "urlTemplate": "alignments.bam",
      "histogram": {
        "storeClass": "JBrowse/Store/SeqFeature/BigWig",
        "urlTemplate": "coverage.bw"
      }
    }
  ]
}

```


One way to command line automate this is to create a file like this called track.json

```
{
  "label": "bam",
  "urlTemplate": "alignments.bam",
  "histogram": {
    "storeClass": "JBrowse/Store/SeqFeature/BigWig",
    "urlTemplate": "coverage.bw"
  }
}
```

Then use bin/add-track-json.pl like this

     cat track.json |bin/add-track-json.pl trackList.json


This will automatically paste the trackList.json into the tracks array in trackList.json

Lot of info but hope that helps

**Reply by katemortensen** [reply]
This is wonderful, thank you! Things are starting to make more sense. I especially like that last trick!

**Comment by cmdcolin** [comment, marked as answer]
Also note: if you are referring to a file like a gene track loaded from flatfile-to-json.pl, histograms are automatically generated for gene tracks generally and if they are just not showing up that may be a different issue. Feel free to post an issue to https://github.com/GMOD/jbrowse

**Reply by katemortensen** [reply]
Ah ha! I think this is my problem. I'll study the mentioned link. Thanks again!
